servicetalk-loadbalancer/src/main/java/io/servicetalk/loadbalancer/DefaultLoadBalancer.java

```diff
@@ -179,6 +158,38 @@ private void subscribeToEvents(boolean resubscribe) {
         }
     }
 
+    // This method is called eagerly, meaning the completable will be immediately subscribed to,
+    // so we don't need to do any Completable.defer business.
+    private Completable doClose(final boolean graceful) {
+        CompletableSource.Processor processor = Processors.newCompletableProcessor();
+        sequentialExecutor.execute(() -> {
+            try {
+                if (!isClosed) {
+                    discoveryCancellable.cancel();
+                    eventStreamProcessor.onComplete();
```

Comment:
Why these 2 lines protected by `if (!isClosed)`?
`toAsyncCloseable` has a CAS internally to make sure it's executed only once.

Reply:
I believe in `toAsyncClosable` the `ClosableResource` can be called twice: once for gracefully and then again for a hard close.